cli/tools/upgrade.rs

```diff
@@ -11,16 +13,100 @@ use deno_runtime::deno_fetch::reqwest::Client;
 use once_cell::sync::Lazy;
 use std::env;
 use std::fs;
+use std::io;
+use std::io::Read;
+use std::io::Seek;
 use std::io::Write;
 use std::path::Path;
 use std::path::PathBuf;
 use std::process::Command;
+use std::time::Duration;
+use std::time::SystemTime;
 
 static ARCHIVE_NAME: Lazy<String> =
   Lazy::new(|| format!("deno-{}.zip", env!("TARGET")));
 
 const RELEASE_URL: &str = "https://github.com/denoland/deno/releases";
 
+const UPGRADE_CHECK_INTERVAL: Duration = Duration::from_secs(12 * 60 * 60);
+
+const UPGRADE_CHECK_FETCH_DELAY: Duration = Duration::from_millis(500);
+
+pub fn check_for_upgrades() -> Option<String> {
+  let current_exe_mtime =
+    env::current_exe().ok()?.metadata().ok()?.modified().ok()?;
```

Comment:
Discussed offline with David, seems we could just store latest version information in the file and compare it against currently running binary version. It would also require fewer syscalls than querying for metadata of the file (mtime).